packages/autorest.typescript/src/generators/clientFileGenerator.ts

```diff
@@ -647,11 +654,8 @@ function getTrack2DefaultContent(
   packageDetails: PackageDetails,
   clientDetails: ClientDetails
 ) {
-  const {
-    azureArm,
-    allowInsecureConnection,
-    addCredentials
-  } = getAutorestOptions();
+  const { azureArm, allowInsecureConnection } = getAutorestOptions();
+  const { addCredentials } = getSecurityInfoFromModel(clientDetails.security);
```

Comment:
Here is the only change for this pr, others are formattings. We get the option `addCredentials` from `getSecurityInfoFromModel` not `getAutorestOptions`.

Reply:
I remember there's a case where we set add-credentials true for future manual customization and don't have a security schema defined.

Reply:
Currently fix we would ignore the credential info if there is security schema defined. If we'd like to support this case we need to decide how to generate the code do you have more info for that case?